plugin/src/main/java/org/opensearch/ml/model/MLModelGroupManager.java

```diff
@@ -101,9 +122,16 @@ public void createModelGroup(MLRegisterModelGroupInput input, ActionListener<Str
                     } else {
                         MLModelGroup.MLModelGroupBuilder builder = MLModelGroup.builder();
                         MLModelGroup mlModelGroup;
+
+                        // TODO: Remove security-related entries from MLModelGroup builder
                         if (modelAccessControlHelper.isSecurityEnabledAndModelAccessControlEnabled(user)) {
                             validateRequestForAccessControl(input, user);
                             builder = builder.access(input.getModelAccessMode().getValue());
+
+                            // share with current user's backend-roles
+                            // TODO: check if resource should be shared with user's backend roles by default
+                            recipientMap.set(Map.of(Recipient.BACKEND_ROLES, Set.copyOf(user.getBackendRoles())));
```

Comment:
I am little confused here. Why are we taking all backend roles here. What if user had 3 backend and is sharing the resource only with one of his backend roles? In that case, shouldn't we use `input.getBackendRoles()`? In the create model group request, user can give what backend roles to share with

Reply:
That's why I had added todo above. I'm not completely aware of the current behavior. Ideally, by default resource shouldn't be shared with anyone upon creation.